modules/carousel/Carousel.tsx

```diff
@@ -222,79 +248,106 @@ const Carousel: React.FC<ICarouselProps & WithStylesProps<Theme, ICarouselStyles
     addEvent("mouseleave", scrollableElementRef?.current, () =>
       onMouseUp(scrollableElementRef?.current)
     );
-  }, [scrollableElementRef]);
+  }, [blockScroll, scrollableElementRef]);
 
   const calculateChildrenSize = useCallback<() => number>(() => {
     if (!scrollableElementRef?.current) {
       return 0;
     }
 
-    const element: HTMLElement = scrollableElementRef.current;
+    const element = scrollableElementRef.current.children[scrollType === "centered" ? 1 : 0];
 
     return (
-      (element.children[0]?.clientWidth || 0) +
+      (element?.clientWidth || 0) +
```

Comment:
If we used `offsetWidth `, wouldn't the extra calcs to get the borders be unnecessary? Maybe we could even get rid of the `borderWidth` prop?